hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/action/clean/CleanPlanner.java

```diff
@@ -510,7 +510,7 @@ public Option<HoodieInstant> getEarliestCommitToRetain() {
       }
     } else if (config.getCleanerPolicy() == HoodieCleaningPolicy.KEEP_LATEST_BY_HOURS) {
       Instant instant = Instant.now();
-      ZonedDateTime currentDateTime = ZonedDateTime.ofInstant(instant, ZoneId.systemDefault());
+      ZonedDateTime currentDateTime = ZonedDateTime.ofInstant(instant, HoodieInstantTimeGenerator.getTimelineTimeZone().getZoneId());
       String earliestTimeToRetain = HoodieActiveTimeline.formatDate(Date.from(currentDateTime.minusHours(hoursRetained).toInstant()));
```

Comment:
The code only works when `HoodieInstantTimeGenerator.setCommitTimeZone` is executed in the same JVM process, we must asure that. Either set it up explicitly or instantiate a `HoodieTableConfig` within which the timezone is set up.

Reply:
tableConfig is generated for the first time when table is created and the time zone is set right then. after that, we can't alter the time zone. 
This is in line w/ how we generate our commit times as well. 

not sure if we need any fixes here. can you help clarify

Reply:
> tableConfig is generated for the first time when table is created

Can we always ensure table config is initialized first? How to guard this sequence dependency.

Reply:
I see the issue. may be we should not set any default value for commitTimeZone in HoodieInstantTimeGenerator. we can set it to null infact. And so unless the table properties are instantiation whcih in turn will call into setCommitTimeZone, if any other callers tries to access the commit time zone, will fail fast.

Reply:
I made it an option in the latest commit but I see some failures in Azure CI where option value is not present for timeline timezone. Its in hudi-cli and `HoodieJavaGenerateApp`, both of these use `HoodieInstantTimeGenerator` directly without setting timezone.

Reply:
Yeah, that means the code is prune to making misusages, let's fix all those test falures by initialzing the zoneId manually.

Reply:
I have changed the PR to use metaClient's tableConfig. Please take a look.
I think `HoodieInstantTimeGenerator` usage needs some refactoring. The static access doesn't seem right.

Reply:
> I think HoodieInstantTimeGenerator usage needs some refactoring

Yes, maybe we should pass in the table path as param and triggers a lazy initialization for the zoneId if necessary.

Reply:
Yeah. We can handle it in a separate jira though.